presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java

```diff
@@ -2945,21 +2995,15 @@ else if (context.getDriverInstanceCount().isPresent()) {
             return new PhysicalOperation(new LocalExchangeSourceOperatorFactory(context.getNextOperatorId(), node.getId(), localExchangeFactory), makeLayout(node), context, exchangeSourcePipelineExecutionStrategy);
         }
 
-        @Override
-        public PhysicalOperation visitNativeExecution(NativeExecutionNode node, LocalExecutionPlanContext context)
-        {
-            OperatorFactory operatorFactory = new NativeExecutionOperator.NativeExecutionOperatorFactory(
-                    context.getNextOperatorId(),
-                    node.getId(),
-                    fragment.withSubPlan(node.getSubPlan()),
-                    context.getTableWriteInfo(),
-                    new PagesSerdeFactory(blockEncodingSerde, isExchangeCompressionEnabled(session), isExchangeChecksumEnabled(session)));
-            return new PhysicalOperation(operatorFactory, makeLayout(node), context, UNGROUPED_EXECUTION);
-        }
-
         @Override
         public PhysicalOperation visitPlan(PlanNode node, LocalExecutionPlanContext context)
         {
+            for (CustomVisitor visitor : customPlanVisitors) {
```

Comment:
I'd like to think more about the CustomVisitor.  This abstraction works for NativeExecutionNode, but makes some assumptions about the custom nodes.  e.g. what if I had a tree as below.  

```
CustomNodeTypeA
       |
CustomNodeTypeB
      |
Regular non-custom node/tree (e.g. TableScanNode)
```

How would the customNodeTypeA translator call the other custom visitors to visit its source node CustomNodeTypeB, which it doesn't understand? and how would CustomNodeTypeB visitor call the regular LocalExecutionPlanner visitor to visit its source, which is a regular plannode?

Reply:
@rschlussel Since we passed the LocalExecutionPlanner visitor as an argument to the CustomVisitor's `translate` method, the user can use that to visit Regular non-custom node inside the CustomVisitor. For CustomNodeTypeB I was thinking the user needs to provide a second CustomVisitor to translate that (since we have List<CustomVisitor> here). Let me add some test cases to show how do we handle these two scenarios.